### PR TITLE
Update Slack URL

### LINKED ISSUE #2009: Incorrect Slack URL?
A URL for `http://slack.writethedocs.org` appears in various texts on the site. You can find occurrences by going to the directory root and running this command: 

```bash
grep -ir "slack\.writethedocs\.org" .
```

As far as I can tell, this link is broken and should be updated to `https://writethedocs.slack.com`. I'm happy to make the fix myself, but I just wanted to make the issue first in case I'm missing something.

### PR DESCRIPTION
**Pull request by MattDodsonEnglish**
Fixes #2009,
changing the URL from  `http://slack.writethedocs.org` to `https://writethedocs.org/slack`.

I changed these with something like:

```shell
find docs/ -type f \( -iname \*.rst -o -iname \*.html -o -iname \*.md \) \
    -exec sed -i \
    "s,http://slack\.writethedocs\.org,https://writethedocs\.org/slack,g" {} + 

```

It changes about 150 lines, but I've tried to check each diff.

### CODE REVIEW

**Comment by MattDodsonEnglish** [comment]
After @ericholscher 's comment in #2009

> slack.writethedocs.org used to be a web app that let you register for Slack, and then a redirect to https://www.writethedocs.org/slack/, which is likely where most of those links should point.

I've updated the links to point to this page on the WTD site (rather than directly to the slack app) with [4bb8ccc](https://github.com/writethedocs/www/pull/2010/commits/4bb8ccc8d93468f6eb495731f4aeade380d51cdd).

**Comment by plaindocs** [comment]
This is great, thank you! 

Because you've touched so many files, the linter is complaining about headings (mostly). 
I'm happy to merge this and ignore those, or wait for either you or me to fix them, as you like.

**Comment by MattDodsonEnglish** [comment]
> I'm happy to merge this and ignore those, or wait for either you or me to fix them, as you like.

If it were a personal project, I'd fix the style issues in a separate PR, just to contain the scope and footprint of a commit. But that's really a matter of Git preferences―I leave that decision to the maintainers :-). Maybe you all also prefer that every merge passes the CI checks.

I don't mind making the style fixes at all, though. I also should probably update the [WTD style page](https://github.com/writethedocs/www/blob/main/docs/style-guide.rst#naming-conventions-titles) , too. It currently recommends title case, putting it at odds with the linter rules:

> For page titles, or Heading1 as they are sometimes called, we use title-case.

So just let me know! Do you prefer I change the style in this PR or another? And should I update the style page?

**Comment by plaindocs** [comment]
The deployment is subject to `main` passing, so we're definitely not merging before the checks pass.

What I was getting at is that just because you fixed the slack issue, I don't necessarily expect you to fix the linting too, although it would be lovely if you did. 

If so, just push them up into this PR. Including fixes to the style guide. style guide out of sync with the linter is pretty bad!

**Comment by MattDodsonEnglish** [comment]
> If so, just push them up into this PR. Including fixes to the style guide. style guide out of sync with the linter is pretty bad!

No problem, I can make this fix. I just need to block out a bit of time to look at. I think I can some time in the next day or so.

**Comment by MattDodsonEnglish** [comment]
@plaindocs I'm not sure why, but some of the linting errors were for files I didn't touch. Anyway, I think I've fixed them all.

In maintenance, every fix brings new fixes, so I also needed to add few more exceptions to the headings rule. I'm not sure about how to capitalize _Docs as [Cc]ode_, so I just left the capitalization as it's written in the page and added an exception. Like a doctor, I operate by the principle _first do no harm_, so I didn't want to editorialize :slightly_smiling_face:  .

  Maybe in another PR I can do a broad linting sweep for other rules.

**Comment by MattDodsonEnglish** [comment]
@plaindocs Friendly reminder about this, when you have a chance :slightly_smiling_face: .

File: docs/guide/api/apiblueprint.rst [hunk 1]
```diff
@@ -1,18 +1,18 @@
 =============
-API Blueprint
+API blueprint
```
Comment (plaindocs) [inline docs/guide/api/apiblueprint.rst]: This one should be "B" I think

Comment (plaindocs) [reply docs/guide/api/apiblueprint.rst]: The exception would have to be API Blueprint, not Blueprint?

**Review by plaindocs — APPROVED** [review]
This is great! 

Thanks for the thoughtful commit stream, made it much easier to review than the scary number of affected files made it look.

File: docs/guide/api/apiblueprint.rst (hunk 1 above)
Comment (MattDodsonEnglish) [reply docs/guide/api/apiblueprint.rst]: Thanks for the careful review! I didn't even realize that "API Blueprint" was a proper noun. 

Blueprint was already an exception, but that file had other title case headings so I over corrected. 

Now I've fixed the heading, and updated the exception.

**Pull request merged**